src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java

```diff
@@ -49,8 +49,11 @@ public Film getFilmById(@PathVariable("id") Integer filmId) {
     }
 
     @GetMapping("/popular")
-    public List<Film> getTopFilms(@Positive @RequestParam(value = "count", defaultValue = "10") Integer count) {
-        return filmService.getTopFilms(count);
+    public List<Film> getTopFilms(
+            @Positive @RequestParam(value = "count", defaultValue = "10", required = false) Integer count,
```

Comment:
Когда есть значение по [умолчанию](https://www.baeldung.com/spring-request-param#:~:text=5.%20%D0%97%D0%BD%D0%B0%D1%87%D0%B5%D0%BD%D0%B8%D0%B5%20%D0%BF%D0%BE%20%D1%83%D0%BC%D0%BE%D0%BB%D1%87%D0%B0%D0%BD%D0%B8%D1%8E%20%D0%B4%D0%BB%D1%8F%20%D0%BF%D0%B0%D1%80%D0%B0%D0%BC%D0%B5%D1%82%D1%80%D0%B0%20%D0%B7%D0%B0%D0%BF%D1%80%D0%BE%D1%81%D0%B0)  `required = false` излишний.


```suggestion
            @Positive @RequestParam(value = "count", defaultValue = "10") Integer count,
```